projects/buttons_and_adc/src/main.c

```diff
@@ -0,0 +1,56 @@
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress s_adc_pin_address = { .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress s_button_address = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings s_button_settings = {
+  .direction = GPIO_DIR_IN,         //
+  .resistor = GPIO_RES_NONE,        //
+  .alt_function = GPIO_ALTFN_NONE,  //
+};
+
+static GpioSettings s_adc_pin_settings = {
+  .direction = GPIO_DIR_IN,           //
+  .resistor = GPIO_RES_NONE,          //
+  .alt_function = GPIO_ALTFN_ANALOG,  //
+};
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,       //
+  .priority = INTERRUPT_PRIORITY_NORMAL,  //
+};
+
+static void prv_callback(const GpioAddress *address, void *context) {
+  uint16_t reading = 0;
+  GpioAddress *s_adc_pin_address = context;
+
+  adc_read_converted_pin(*s_adc_pin_address, &reading);
```

Comment:
Here `s_adc_pin_address` is a local variable, so it shouldn't be prefixed with `s_`; furthermore, it's shadowing the global `s_adc_pin_address`, so it definitely needs a different name.